app/steps/medicaid/insurance_current_type.rb

```diff
@@ -2,6 +2,41 @@
 
 module Medicaid
   class InsuranceCurrentType < Step
-    step_attributes(:insurance_type)
+    step_attributes(
+      :insurance_type,
+      :members,
+    )
+
+    def insured_members_requesting_insurance
+      members.select(&:requesting_health_insurance).select(&:is_insured)
+    end
+
+    def valid?
+      if requesting_members_valid?
+        true
+      else
+        errors.add(
+          :insurance_type,
+          error_message,
+        )
+        false
+      end
+    end
+
+    private
+
+    def requesting_members_valid?
+      insured_members_requesting_insurance.all? do |m|
```

Comment:
this seems similar to `ManyMembersStep`, may consider inheriting from that

Reply:
✅